apps/bfd-pipeline/bfd-pipeline-rda-grpc/src/main/java/gov/cms/bfd/pipeline/rda/grpc/sink/direct/AbstractClaimRdaSink.java

```diff
@@ -397,16 +400,24 @@ private long maxSequenceInBatch(Collection<RdaChange<TClaim>> claims) {
   }
 
   /**
-   * Updates the latency metric by adding the age of each claim to the histogram.
+   * Updates the latency metrics by adding the age of each claim to the histograms.
    *
    * @param claims claims to process
    */
-  private void updateLatencyMetric(Collection<RdaChange<TClaim>> claims) {
+  private void updateLatencyMetrics(Collection<RdaChange<TClaim>> claims) {
+    final long nowMillis = clock.millis();
     for (RdaChange<TClaim> claim : claims) {
-      final long nowMillis = clock.millis();
       final long changeMillis = claim.getTimestamp().toEpochMilli();
-      final long age = Math.max(0L, nowMillis - changeMillis);
-      metrics.changeAgeMillis.update(age);
+      final long changeAge = Math.max(0L, nowMillis - changeMillis);
+      metrics.changeAgeMillis.update(changeAge);
+
+      final LocalDate extractDate = claim.getSource().getExtractDate();
+      if (extractDate != null) {
+        final LocalDateTime extractTime = extractDate.atStartOfDay();
+        final long extractMillis = extractTime.toInstant(ZoneOffset.UTC).toEpochMilli();
+        final long extractAge = Math.max(0L, nowMillis - extractMillis);
+        metrics.extractAgeMillis.update(extractAge);
```

Comment:
If my math is correct, and we want to set an alarm for data received after 2pm eastern on the following day, we'd put our threshold at 43 hours (or 154,800,000 milliseconds).  I believe that would account for the shift from eastern to UTC.  Is that correct?

Reply:
Both the `extractTime` and the `nowMillis` are in UTC so unless I'm mistaken the difference between them should be a true difference in wall clock time.

Reply:
Just to be safe I have made a small change to ensure that the `nowMillis` and `extractTime` are both computed using the same time zone.  This way even if the clock passed to the sink's constructor is not in UTC the relative difference will still be the same.